sysbuild/CMakeLists.txt

```diff
@@ -52,38 +94,144 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
   endif()
 
   if(SB_CONFIG_BOOTLOADER_MCUBOOT)
-    set_config_string(${DEFAULT_IMAGE} CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS
-                      "--pad-header --header-size ${SB_CONFIG_PM_MCUBOOT_PAD}"
-    )
+    # Make mcuboot a build only target as the main application will flash this from the
+    # merged hex file
+    set_target_properties(mcuboot PROPERTIES BUILD_ONLY true)
+    set_property(TARGET mcuboot APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES}\n")
+    set_property(TARGET ${DEFAULT_IMAGE} APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES}\n")
+
+    # Set MCUboot operating mode in application and MCUboot
+# todo: support RAM load
+    set(mcuboot_mode)
+    set(application_mcuboot_mode)
```

Comment:
what is the purpose of having these empty assignments?

Reply:
Removed